irods/exception.py

```diff
@@ -67,8 +69,60 @@ def __init__(self, name, bases, attrs):
             iRODSExceptionMeta.codes[attrs['code']] = self
 
 
+class Errno:
+    """Encapsulates an integer error code from the operating system
+       and provides a text representation.
+    """
+    def __init__(self,arg0,*args):
+        """Initializes an object with an integer error code arg0.
+           Further arguments are optional and ignored.
+        """
+        self.int_code = arg0
+    def __repr__(self):
+        e = self.int_code
+        try:
+            return self.__class__.__name__ + repr(tuple([e, errno.errorcode[e], os.strerror(e)]))
+        except Exception:
+            pass
+        return self.__class__.__name__ + repr(tuple([e,]))
+    def __int__(self):
+        return self.int_code
+
+
 class iRODSException(six.with_metaclass(iRODSExceptionMeta, Exception)):
-    pass
+    """An exception that originates from a server error.
+       Exception classes that are derived from this base and represent a concrete error, should
+       store a unique error code (X*1000) in their 'code' attribute, where X < 0.
```

Comment:
Should we enforce the `x < 0` constraint?
Is it acceptable to store a positive integer in this type?

Reply:
Will test this combined with the other PR's this morning. Then maybe, release.